core/src/main/java/org/infinispan/container/offheap/BoundedOffHeapDataContainer.java

```diff
@@ -155,8 +155,10 @@ public void addSegments(IntSet segments) {
       // Don't have to do anything here
    }
 
+   private static final int REMOVE_SEGMENT_BATCH_SIZE = 32;
+
    @Override
    public void removeSegments(IntSet segments) {
-      InternalDataContainerAdapter.removeSegmentEntries(this, segments, listeners, keyPartitioner);
+      NonSegmentedUtils.removeSegmentEntries(this, segments, listeners, keyPartitioner);
```

Comment:
Couldn't we throw an UnsupportedOperationException instead?

Reply:
I would like to, but until we remove the segmentation feature flag, we have to keep this for now.